setup/command/pull.go

```diff
@@ -33,3 +62,32 @@ func (f *CommandFactory) Pull() cli.Command {
 		},
 	}
 }
+
+func getRemoteInstanceBucket(s s3iface.S3API, instanceName string) (string, error) {
+	listBucketsOutput, err := s.ListBuckets(&s3.ListBucketsInput{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, bucket := range listBucketsOutput.Buckets {
+		bucketName := aws.StringValue(bucket.Name)
+
+		if split := strings.Split(bucketName, "-"); len(split) == 3 && split[0] == "layer0" && split[1] == instanceName {
+			return bucketName, nil
+		}
+	}
+
+	return "", fmt.Errorf("No S3 bucket found for given instance name")
```

Comment:
[nit] should probably show the instance name with some quotes here: `...No S3 bucket found for instance '%s'", instanceName)`